src/main/scala/uk/gov/nationalarchives/tdr/api/db/repository/FileRepository.scala

```diff
@@ -58,6 +58,17 @@ class FileRepository(db: Database)(implicit val executionContext: ExecutionConte
     db.run(query.result)
   }
 
+  def countFilesOrFoldersInConsignment(consignmentId: UUID,
```

Comment:
I think it would be safe to change the existing `countFilesInConsignment` method, as below and use this for the pagination counts:

```
def countFilesInConsignment(consignmentId: UUID,
                              parentId: Option[UUID] = None,
                              fileTypeIdentifier: Option[String] = Some(NodeType.fileTypeIdentifier)): Future[Int] = {
    val query = File.filter(_.consignmentid === consignmentId)
      .filterOpt(fileTypeIdentifier)(_.filetype === _)
      .filterOpt(parentId)(_.parentid === _)
      .length
    db.run(query.result)
  }
```
We want to try and avoid creating more methods than are needed, as we already have a lot, doing very similar things. You can re-use the existing unit tests you'd written for the new count method, just including a new one where the `fileTypeIdentifier` is passed in as `None`.

Also I think for the pagination count I think you'll want to get back all the direct children nodes for a given parent id, to provide the correct count to the UI for the display, which would mean using a `fileTypeIdentifier` of `None`

Reply:
This makes sense. The `getPaginatedFiles` method currently calls the `countFilesInConsignment` method with a parameter which could be `None`. So the default value won't be used.